### PR TITLE
[Fix] dots1 expert bias routing

### PR DESCRIPTION
**Pull request by fjosw**
# What does this PR do?

This is a follow up to https://github.com/huggingface/transformers/pull/41647. I went over the model code again and noticed that the dots1 routing apparently also broke in https://github.com/huggingface/transformers/pull/40132. dots1 seems to be very close in architecture to Deepseek V3 but not explcitly dependent. Not sure if there is a more elegant way to fix this?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@ArthurZucker

### CODE REVIEW

**Comment by fjosw** [comment]
Reminder: I think the dots1 model is still broken on main @ArthurZucker, not sure if this PR is the best way to address this, though.

**Review by vasqu — APPROVED** [review]
LGTM, I double checked and this looks correct to me. The selection is affected by the e_score_bias but not the final logits that are then selected. 

Imo, we could refactor this later on to be within the router instead! (including deepseek). cc @3outeille since you worked a lot on MoEs recently it might be interesting
